### DISCUSSION TITLE
Use of third party controls (i.e. Atalasoft DotImage) with Avalonia?

### DISCUSSION THREAD

**Discussion by drabina** [AvaloniaUI/Avalonia#16902] [General]
I am looking to create a roadmap to update all our applications from WinForms to a more modern UI. With all the mess that Microsoft created with all their UIs, Avalonia looks like a perfect solution. My only concern is support for third party controls. We use Atalasoft extensively (DotImage, DotTwain, etc) and would need to retain it in our apps. Does Avalonia support third party controls? I know that Atalasoft is compatible with WPF but I am not sure if I could use their controls in Avalonia.

**Comment by colejohnson66** [comment]
Avalonia supports third-party controls, *as long as they are Avalonia controls.* If you *need* WPF controls, take a look at Avalonia XPF.
